src/pages/IstioConfigList/FiltersAndSorts.ts

```diff
@@ -10,28 +10,53 @@ export namespace IstioConfigListFilters {
       title: 'Namespace',
       isNumeric: false,
       param: 'ns',
-      compare: (a: IstioConfigItem, b: IstioConfigItem) => 1
+      compare: (a: IstioConfigItem, b: IstioConfigItem) => {
+        return a.namespace.localeCompare(b.namespace) || a.name.localeCompare(b.name);
+      }
     },
     {
       id: 'istiotype',
       title: 'Istio Type',
       isNumeric: false,
       param: 'it',
-      compare: (a: IstioConfigItem, b: IstioConfigItem) => 1
+      compare: (a: IstioConfigItem, b: IstioConfigItem) => {
+        return a.type.localeCompare(b.type) || a.name.localeCompare(b.name);
+      }
     },
     {
       id: 'istioname',
       title: 'Istio Name',
       isNumeric: false,
       param: 'in',
-      compare: (a: IstioConfigItem, b: IstioConfigItem) => 1
+      compare: (a: IstioConfigItem, b: IstioConfigItem) => a.name.localeCompare(b.name)
     },
     {
       id: 'configvalidation',
       title: 'Config',
       isNumeric: false,
       param: 'cv',
-      compare: (a: IstioConfigItem, b: IstioConfigItem) => 1
+      compare: (a: IstioConfigItem, b: IstioConfigItem) => {
+        let sortValue = -1;
+        if (a.validation && !b.validation) {
+          sortValue = -1;
+        } else if (!a.validation && b.validation) {
+          sortValue = 1;
+        } else if (!a.validation && !b.validation) {
+          sortValue = 0;
+        } else if (a.validation && b.validation) {
+          if (a.validation.valid && !b.validation.valid) {
+            sortValue = -1;
+          } else if (!a.validation.valid && b.validation.valid) {
+            sortValue = 1;
+          } else if (a.validation.valid && b.validation.valid) {
```

Comment:
If I understand correctly, this means: if both are valid, order with the one that has more checks first. If both are invalid, do the opposite. Correct?

I'm not really against that but I wonder if it really brings anything to have such complex logic. Maybe here, having sortValue=0 (hence fallback to name sorting) would be enough? As you wish, just suggesting.

Reply:
Oh actually, I'm just seeing this isn't (entirely) new, it's just moving an existing piece of code - except that in this piece of code, when both are valid (not invalid) we were doing sortValue=0. I guess then we should keep it as before?

Reply:
yep, I didn't want to enter into refactoring the logic itself, just adapting that code to the new shape of sortings. However, Hayk found a bug that affects to the existent logic (master branch has it), so I use the JIRA to fix it.
The reason of changing that line is because you can have to valid items with checks in it. So, you'll need to place first valid items without checks, then valid items with checks and then invalid items.

Reply:
It would probably be easier if we had the 'valid' field as a non-binary attribute (i.e.: valid, warning, error). This way, the comparative would be quite trivial.